core/src/main/scala/org/apache/spark/eventhubs/client/ClientConnectionPool.scala

```diff
@@ -69,41 +69,42 @@ object ClientConnectionPool extends Logging {
 
   private[this] val pools = new MutableMap[String, ClientConnectionPool]()
 
-  def isInitialized(name: String): Boolean = pools.synchronized {
-    pools.get(name).isDefined
+  def isInitialized(key: String): Boolean = pools.synchronized {
+    pools.get(key).isDefined
   }
 
-  private def ensureInitialized(name: String): Unit = {
-    if (!isInitialized(name)) {
-      val message = notInitializedMessage(name)
+  private def ensureInitialized(key: String): Unit = {
+    if (!isInitialized(key)) {
+      val message = notInitializedMessage(key)
       throw new IllegalStateException(message)
     }
   }
 
-  private def get(name: String): ClientConnectionPool = pools.synchronized {
-    pools.getOrElse(name, {
-      val message = notInitializedMessage(name)
+  private def key(ehConf: EventHubsConf): String = {
+    ehConf.connectionString
```

Comment:
consider normalizing the `connectionString`
- to handle the case where customers could give any `case` on `namespace` and `eventhub names` and pack them in any order into the connection string.

Probably, having this normalized while initializing `EHConf` is the way-to-go - to eliminate the need to validate this on every call.

Reply:
I agree, I will add some normalization to both keys!